framework/framework.go

```diff
@@ -142,3 +142,60 @@ func (f *Framework) ProcessDelete(obj interface{}, resources []Resource) error {
 
 	return nil
 }
+
+// ProcessUpdate is a drop-in for an informer's UpdateFunc. It receives the
+// custom object observed during TPR watches and anything that implements
+// Resource. ProcessUpdate takes care about all necessary reconciliation logic
+// for update events. For complex resources this means state has to be created,
+// deleted and updated eventually, in this order.
+//
+//     func updateFunc(oldObj, newObj interface{}) {
+//         err := f.ProcessUpdate(oldObj, newObj, resources)
+//         if err != nil {
+//             // error handling here
+//         }
+//     }
+//
+//     newResourceEventHandler := &cache.ResourceEventHandlerFuncs{
+//         UpdateFunc:    updateFunc,
+//     }
+//
+func (f *Framework) ProcessUpdate(oldObj, newObj interface{}, resources []Resource) error {
```

Comment:
> We could argue that it does not make sense to have OnAdd and OnUpdate

Naming isn't that important for me atm. We could remove `OnUpdate` and leave `OnAdd`. It's the same thing for me.

The handler godoc is one thing, and actual implementation is another. IIRC both `OnAdd` and  `OnUpdate` are called when TPO changes. I think in the long run we may want to switch from `cache` package to pure http streaming as most of the operators do. If we have both those methods this may become difficult.

> OnUpdate implies OnAdd, but this is not in the scope of the operatorkit reconciliation framework.

I think it is because if we have only one of those methods, it enforces users to write (more) correct reconciliation. And I believe this is what this framework is all about.